src/utils.rs

```diff
@@ -232,22 +232,19 @@ pub fn utc_datetime_from_epoch_ms(millis: u64) -> DateTime<Utc> {
 pub fn into_jsrpc_error(err: Web3Error) -> Error {
     Error {
         code: match err {
-            Web3Error::InternalError | Web3Error::NotImplemented => ErrorCode::InternalError,
+            Web3Error::InternalError(_) | Web3Error::NotImplemented => ErrorCode::InternalError,
             Web3Error::NoBlock
             | Web3Error::PrunedBlock(_)
             | Web3Error::PrunedL1Batch(_)
-            | Web3Error::NoSuchFunction
-            | Web3Error::RLPError(_)
-            | Web3Error::InvalidTransactionData(_)
+            | Web3Error::ProxyError(_)
             | Web3Error::TooManyTopics
             | Web3Error::FilterNotFound
-            | Web3Error::InvalidFeeParams(_)
             | Web3Error::LogsLimitExceeded(_, _, _)
-            | Web3Error::InvalidFilterBlockHash => ErrorCode::InvalidParams,
-            Web3Error::SubmitTransactionError(_, _) | Web3Error::SerializationError(_) => 3.into(),
-            Web3Error::PubSubTimeout => 4.into(),
-            Web3Error::RequestTimeout => 5.into(),
-            Web3Error::TreeApiUnavailable => 6.into(),
+            | Web3Error::InvalidFilterBlockHash
+            | Web3Error::TreeApiUnavailable => ErrorCode::InvalidParams,
+            Web3Error::SubmitTransactionError(_, _) | Web3Error::SerializationError(_) => {
+                ErrorCode::ServerError(3)
```

Comment:
Where does the `3` come from?

Reply:
Truly not sure lol we were using it previously (e.g. `3.into()`). It puts it into the `JSONRPC error code` enum as expected. Happy to change but kept it as it was done previously.

Reply:
Ah yup, let's keep as-is 😅